test/unit/org/apache/cassandra/utils/CassandraGenerators.java

```diff
@@ -413,10 +491,250 @@ public Gen<KeyspaceMetadata> build()
         }
     }
 
+    public static Gen<CachingParams> cachingParamsGen()
+    {
+        return rnd -> {
+            boolean cacheKeys = nextBoolean(rnd);
+            int rowsPerPartitionToCache;
+            switch (SourceDSL.integers().between(1, 3).generate(rnd))
+            {
+                case 1: // ALL
+                    rowsPerPartitionToCache = Integer.MAX_VALUE;
+                    break;
+                case 2: // NONE
+                    rowsPerPartitionToCache = 0;
+                    break;
+                case 3: // num values
+                    rowsPerPartitionToCache = Math.toIntExact(rnd.next(Constraint.between(1, Integer.MAX_VALUE - 1)));
+                    break;
+                default:
+                    throw new AssertionError();
+            }
+            return new CachingParams(cacheKeys, rowsPerPartitionToCache);
+        };
+    }
+
+    public enum KnownCompactionAlgo
+    {
+        SizeTiered(SizeTieredCompactionStrategy.class),
+        Leveled(LeveledCompactionStrategy.class),
+        Unified(UnifiedCompactionStrategy.class);
```

Comment:
No TWCS?

Reply:
so thought about this but i felt that the "time" part would make tests non-deterministic